src/ray/rpc/server_call.h

```diff
@@ -194,21 +204,54 @@ class ServerCallImpl : public ServerCall {
   void SetState(const ServerCallState &new_state) override { state_ = new_state; }
 
   void HandleRequest() override {
+    bool auth_success = true;
+    if constexpr (EnableAuth == AuthType::STRICT_AUTH) {
+      RAY_CHECK(!cluster_id_.IsNil()) << "Expected cluster ID in server call!";
+      auto &metadata = context_.client_metadata();
+      if (auto it = metadata.find(kClusterIdKey);
+          it == metadata.end() || it->second != cluster_id_.Hex()) {
+        RAY_LOG(DEBUG) << "Wrong cluster ID token in request! Expected: "
+                       << cluster_id_.Hex() << ", but got: "
+                       << (it == metadata.end() ? "No token!" : it->second);
+        auth_success = false;
+      }
+    } else if constexpr (EnableAuth == AuthType::LAZY_AUTH) {
+      RAY_CHECK(!cluster_id_.IsNil()) << "Expected cluster ID in server call!";
+      auto &metadata = context_.client_metadata();
+      if (auto it = metadata.find(kClusterIdKey);
+          it != metadata.end() && it->second != cluster_id_.Hex()) {
+        RAY_LOG(DEBUG) << "Wrong cluster ID token in request! Expected: "
+                       << cluster_id_.Hex() << ", but got: "
+                       << (it == metadata.end() ? "No token!" : it->second);
+        auth_success = false;
+      }
+    } else {
+      if (!cluster_id_.IsNil()) {
+        RAY_LOG_EVERY_N(WARNING, 100)
+            << "Unexpected cluster ID in server call! " << cluster_id_;
```

Comment:
Maybe change it to RAY_LOG_EVERY_MS  and give it 5s ? I feel if something goes wrong, this logs might be everywhere.